[Java] Refactor FormatUtil class

Bill7zz opened the pull request:
- Import and update the methods from `DateTimeFormatUtil` to `FormatUtil` class keeping them with the proper implementation. 
- Remove duplicated `DateTimeFormatUtil` class
- Refactor `FormatUtil` class name to `DateTimeFormatUtil` to keep the same pattern used in other languages

Build and tests passed correctly
![image](https://user-images.githubusercontent.com/37461749/51702374-69422500-1ff2-11e9-9496-2a64e0afe9c4.png)

gasper-az reviewed (approved):
LGTM

JuanAr commented on Java/linting-rules.xml:
```diff
@@ -30,7 +30,7 @@
             <property name="allowNonPrintableEscapes" value="true"/>
         </module>
         <module name="LineLength">
-            <property name="max" value="180"/>
```
Is there a way to format the code with line breaks to pass this validation instead of changing it?

Bill7zz replied:
The line exceeds the limit by one character. The possibles solutions are
- Split the method parameters 
- Remove an indentation tab to keep the same format of the other method calls

Bill7zz replied:
![image](https://user-images.githubusercontent.com/37461749/51708308-5c2d3200-2002-11e9-9ec5-6c470a5f5f1c.png)

JuanAr replied:
Which file is this one?

JuanAr replied:
I think indentation is wrong right now. I used Reformat Code in IntelliJ and it reduced the indentation by one tab (4 spaces). if that fixes the issue, go ahead. If not, add the method argument splitting.
![image](https://user-images.githubusercontent.com/805847/51708786-8af7d800-2003-11e9-889b-26f4a4d8362e.png)

Bill7zz replied:
BaseDateTimePeriodParser.java 
Line 94 is breaking the rule.

JuanAr reviewed (changes requested):
I would prefer to not change the linting rules

Bill7zz committed:
- Import and update the methods from `DateTimeFormatUtil` to `FormatUtil` class keeping them with the proper implementation.
- Remove duplicated `DateTimeFormatUtil` class
- Refactor `FormatUtil` class name to `DateTimeFormatUtil` to keep the same pattern used in other languages
